templates/ts/directives/stop-body-scroll.ts

```diff
@@ -25,17 +44,44 @@ Alpine.directive('stop-body-scroll', (
     // This means we are not displayed on the page (e.g. a parent is display: none), so
     // it means it's likely being used via hiding/showing an element, not adding/removing
     // an element. This directive only supports the adding/removing element use case.
-    // If used by hiding/showing the element, you have to manage the body scroll yourself...
+    // If used by hiding/showing the element, you have to manage the body scroll yourself
+    // or use x-stop-body-scroll-if="condition"
     if (!el.offsetParent) return
+    inc()
+    cleanup(() => dec())
+})
 
-    if (count === 0) {
-        stopBodyScroll()
-    }
-    count += 1
+/**
+ * A directive to stop body scroll while a condition is true.
+ *
+ * The purpose is for things like modals/menus/etc. that are shown/hidden, e.g. using x-show
+ * rather than the element being present or not.
+ *
+ * Usage:
+ *
+ *  <div x-stop-body-scroll-if="open"></div>
+ */
+Alpine.directive('stop-body-scroll-if', (
```

Comment:
New fancy directive! ... for when you want to stop body scroll based on a js condition.

The other one `stop-body-scroll` is for when the element is present in the DOM or not, but many use cases it gets hidden/shown based on a condition, so this directive is more appropriate.